### PR TITLE
Adding option to name a list-type artifact

### PR DESCRIPTION
**Pull request by pediejo**
Added optional functionality to insert a name for a list artifact when saving.

### CODE REVIEW

File: src/python/zquantum/core/utils.py [hunk 1]
```diff
@@ -351,15 +351,16 @@ def load_list(file):
...
     """Save expectation values to a file.
 
     Args:
         array (list): the list to be saved
         file (str or file-like object): the name of the file, or a file-like object
+        artifact_name (str): optional argument to specify the schema name
     """
     dictionary = {}
-    dictionary["schema"] = SCHEMA_VERSION + "-list"
+    dictionary["schema"] = SCHEMA_VERSION + artifact_name + "-list"
```
Comment (mstechly) [inline src/python/zquantum/core/utils.py:363]: I think you want to do have "-" in the schema name.

Consider the following syntax:
`"-".join([SCHEMA_VERSION, artifact_name, "list"])

File: src/python/zquantum/core/utils_test.py [hunk 2]
```diff
@@ -155,6 +155,16 @@ def test_list_io(self):
...
         os.remove("list.json")
 
+    def test_named_list_io(self):
+        # Given
+        initial_list = [0.1, 0.3, -0.3]
+        # When
+        save_list(initial_list, "list.json", "number")
+        loaded_list = load_list("list.json")
+        # Then
+        self.assertListEqual(initial_list, loaded_list)
```
Comment (mstechly) [inline src/python/zquantum/core/utils_test.py:165]: You're not checking if the schema value is properly saved here :)

File: src/python/zquantum/core/utils.py (hunk 1 above)
Comment (pediejo) [reply src/python/zquantum/core/utils.py]: Good spot! Fixed this.

File: src/python/zquantum/core/utils_test.py (hunk 2 above)
Comment (pediejo) [reply src/python/zquantum/core/utils_test.py:166]: Okay, tried to add a check of this.

Comment (mstechly) [reply src/python/zquantum/core/utils_test.py:166]: Instead of defining your own `SCHEMA_VERSION`, you should do:
`from .utils import SCHEMA_VERSION` :)

**Pull request merged**